### PR TITLE
Build unit tests in ReactWindows-Universal solution

### PR DESCRIPTION
**Pull request by jonthysell**
* Added MSO and managed unit tests to ReactWindows-Universal.sln solution
* Enabled unit tests for x86 and x64
* Fixed unit tests to build properly on x86
* Cleaned up unnecessary applyRNPatch parameters
* Added `--msbuildprops [string]` argument to `react-native run-windows`
* Specified run-windows tasks in CI loop to redirect the build intermediate folder to get around storage limitations

Addresses issue #4256

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4285)

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
This soaks up 36 agent slots when it's running, which is bringing the shared pool on dev.azure.com/ms to its knees for all other projects sharing it. Is there some way to stem the bleeding here before it makes every PR this huge? 🤔

**Comment by DHowett-MSFT** [comment]
(fyi: this was filed on vscode-python for a similar reason; https://github.com/microsoft/vscode-python/issues/9939)

**Comment by kmelmon** [comment]
@DHowett-MSFT that's really good info!  Would you mind sharing how you found this out?

**Comment by DHowett-MSFT** [comment]
@kmelmon sure! So, I noticed that our CI jobs in Terminal are backed up a bit, so I jumped to the agent pool through one of Terminal's build runs. It took me to a list of all the jobs scheduled on that pool, and their originating projects. :smile:

**Comment by jonthysell** [comment]
Can we increase the size of the official Microsoft pool? Or have different pools for different Microsoft projects? Looking at our ADO pipelines doesn't tell us anything about being blocked by other projects.

**Comment by jonthysell** [comment]
We spread out to more agents because we don't have enough disk space to build everything on one. So we either need more agents or more space, and more space wasn't an option. I'll disable some of the flavors again, but it's rediculous that Microsoft can't adequately fund its own projects on its own platform.

File: .ado/windows-vs-pr.yml [hunk 1]
```diff
@@ -26,30 +26,31 @@ jobs:
...
+        ArmRelease:
           BuildConfiguration: Release
-          BuildPlatform: arm64
+          BuildPlatform: ARM
+        #Arm64Debug:
+        #  BuildConfiguration: Debug
+        #  BuildPlatform: ARM64
+        #Arm64Release:
+        #  BuildConfiguration: Release
+        #  BuildPlatform: ARM64
```
Comment (kmelmon) [inline .ado/windows-vs-pr.yml]: Realizing this is sub-optimal to not build all flavors for all apps, we can get still get pretty coverage by spreading the build flavors across the different test apps.  It looks like we're not building anything in arm64 flavor.  Is that still not working yet?  If not then it's OK to tackle that later.  The thought was have at least one arm64chk and arm64fre combo across some set of test apps.

File: vnext/local-cli/runWindows/runWindows.js [hunk 2]
```diff
@@ -55,8 +55,16 @@ async function runWindows(config, args, options) {
 
     // Get build/deploy options
     const buildType = deploy.getBuildConfiguration(options);
+    const msBuildProps = build.parseMsBuildProps(options);
```
Comment (kmelmon) [inline vnext/local-cli/runWindows/runWindows.js]: >msBuildProps [](start = 10, length = 12)

nice!

File: .ado/windows-vs-pr.yml (hunk 1 above)
Comment (jonthysell) [reply .ado/windows-vs-pr.yml]: We have to retarget our min windows version before we cane have arm64 apps. #4031

**Pull request merged**